src/transmitter/index.ts

```diff
@@ -114,3 +112,15 @@ async function retryRequest(verb: NeedleHttpVerbs, url: string, payload: object)
 
   return {response, attempt};
 }
+
+function shouldRetryRequest(err: IRequestError, networkErrorMessages: string[], stillHaveRetries: boolean): boolean {
+  if (!stillHaveRetries) {
+    return false;
+  }
+
+  if (err.code === RequestCodeError.ECONNRESET && networkErrorMessages.includes(err.message)) {
```

Comment:
➕ 1️⃣  on the Enum idea, I also think it's a really nice feature to use! However, in this current context I think it doesn't introduce benefits. It gives an extra layer of indirection (maybe you want to check what's the value of the enum?) and is a bit longer to type 😄 What do you think of just keeping it like before `(err.code === 'ECONNRESET')` as it is very explicit and easy to read?